File: frontend/src/types/subtask.ts
export interface Subtask {
    id: number;
    title: string;
    description?: string;
    completed: boolean;
    position: number;
    dueDate?: string;
    estimatedHours?: number;
    assignee?: {
        id: number;
        username: string;
        email?: string;
        avatarUrl?: string;
        displayName?: string;
    };
    createdAt: string;
    updatedAt: string;
}

export interface CreateSubtaskRequest {
    title: string;
    description?: string;
    dueDate?: string;
    estimatedHours?: number;
    assigneeId?: number;
}

export interface UpdateSubtaskRequest {
    title?: string;
    description?: string;
    completed?: boolean;
    position?: number;
    dueDate?: string;
    estimatedHours?: number;
    assigneeId?: number;
} 

File: backend/src/main/java/com/yourapp/model/Subtask.java
package com.yourapp.model;

import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.ToString;
import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonManagedReference;

import jakarta.persistence.*;
import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "subtasks")
public class Subtask {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String title;
    
    private String description;
    
    private boolean completed;
    
    private Integer position;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "parent_task_id")
    @ToString.Exclude
    @EqualsAndHashCode.Exclude
    @JsonBackReference("task-subtasks")
    private Task parentTask;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "assignee_id")
    @ToString.Exclude
    @EqualsAndHashCode.Exclude
    private User assignee;
    
    @Column(name = "created_at")
    private LocalDateTime createdAt;
    
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
    
    @Column(name = "due_date")
    private LocalDateTime dueDate;
    
    @Column(name = "estimated_hours")
    private Integer estimatedHours;
    
    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }
    
    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
} 